config/webpack.config.js

```diff
@@ -236,9 +243,17 @@ module.exports = env => {
       }),
 
       new MiniCssExtractPlugin({
-        filename: !isOptimizedBuild
-          ? '[name].css'
-          : `[name].[contenthash]-${timestamp}.css`,
+        moduleFilename: chunk => {
+          const { name } = chunk;
+          const standardFilename = `${name}.[contenthash]-${timestamp}.css`;
+          const isMedalliaStyleFile = name === vaMedalliaStylesFilename;
+          const isStandardFile = !isMedalliaStyleFile;
+          const showHash = isOptimizedBuild && isStandardFile;
+
+          if (showHash) return standardFilename;
+
+          return environment.isStaging() ? `${name}.css` : standardFilename;
```

Comment:
>> Now that I'm getting a second look at this, it seems like dev would keep the hash in the name (not optimized build, so showHash is falsy). Not sure how much this matters.
> 
> @U-DON, my intent for the code was:
> 
> 1. Determine if it's the Medallia CSS file.
> 1. If not, return our normal filename format.
> 1. If so, determine if it's the staging* environment.
>     - My understanding of the requirements is that I should be checking only for staging. If I should be checking for additional non-prod environments, then I need to update my code.
> 1. If staging, exclude the hash.
> 1. If non-staging, include the hash.

Restarting thread on new changes so the discussion doesn't get lost from an outdated snippet.

If the requirements is only for staging, that's fair. I just wanted to point out that this looks like it woud add hashes for dev, which is a deviation from the original code.

If we want to be explicit about "hash only for Medallia, only in staging", then would it be clearer to write it like this?

```suggestion
          const standardFilename = `${name}.[contenthash]-${timestamp}.css`; 
          const isMedalliaStyleFile = name === vaMedalliaStylesFilename; 
          
          if (isMedalliaStyleFile && environment.isStaging()) return `${name}.css`;
          
          return isOptimizedBuild ? standardFilename : `${name.css}`;
```

Reply:
This looks a lot cleaner. Thanks for the suggestion.